release/scripts/modules/rna_keymap_ui.py

```diff
@@ -165,9 +162,9 @@ def draw_kmi(display_keymaps, kc, km, kmi, layout, level):
 
     # Expanded, additional event settings
     if kmi.show_expanded:
```

Comment:
This is what I think, the old UI logic essentially starts to fall apart and created the "Unexpand Upon Change" behavior.  The code changes after this are mostly visual changes made to make keymap editor look nicer.